algosdk/template.py

```diff
@@ -60,65 +63,72 @@ def get_program(self):
                 "EDMACCEFCzMBCCEGCxIQMwAIIQcPEBA=")
         orig = base64.b64decode(orig)
         offsets = [4, 7, 8, 9, 10, 14, 47, 80]
-        values = [self.max_fee, self.expiry_round, self.ratn, self.ratd,
-                  self.min_pay, self.owner, self.receiver_1, self.receiver_2]
+        values = [self.max_fee, self.expiry_round, self.rat_2,
+                  self.rat_1, self.min_pay, self.owner,
+                  self.receiver_1, self.receiver_2]
         types = [int, int, int, int, int, "address", "address", "address"]
         return inject(orig, offsets, values, types)
 
-    def get_send_funds_transaction(self, amount: int, first_valid, last_valid,
-                                   gh, precise=True):
+    @staticmethod
+    def get_split_funds_transaction(contract, amount: int, fee: int, first_valid,
+                                   last_valid, gh):
         """
         Return a group transactions array which transfers funds according to
         the contract's ratio.
 
         Args:
-            amount (int): amount to be transferred
+            amount (int): total amount to be transferred
+            fee (int): fee per byte
             first_valid (int): first round where the transactions are valid
             gh (str): genesis hash in base64
-            precise (bool, optional): precise treats the case where amount is
-                not perfectly divisible based on the ratio. When set to False,
-                the amount will be divided as close as possible but one
-                address will get slightly more. When True, an error will be
-                raised. Defaults to True.
 
         Returns:
             Transaction[]
-
-        Raises:
-            NotDivisibleError: see precise
         """
+        address = logic.address(contract)
+        _, ints, bytearrays = logic.read_program(contract)
+        if not (len(ints) == 8 and len(bytearrays) == 3):
+            raise error.WrongContractError("split")
+        rat_1 = ints[6]
+        rat_2 = ints[5]
+        min_pay = ints[7]
+        max_fee = ints[1]
+        receiver_1 = encoding.encode_address(bytearrays[1])
+        receiver_2 = encoding.encode_address(bytearrays[2])
+
         amt_1 = 0
         amt_2 = 0
 
-        gcd = math.gcd(self.ratn, self.ratd)
-        ratn = self.ratn // gcd
-        ratd = self.ratd // gcd
+        gcd = math.gcd(rat_1, rat_2)
+        rat_1 = rat_1 // gcd
+        rat_2 = rat_2 // gcd
 
-        if amount % ratd == 0:
-            amt_1 = amount // ratd * ratn
-            amt_2 = amount - amt_2
-        elif precise:
-            raise error.NotDivisibleError
-        else:
-            amt_1 = round(amount / ratd * ratn)
+        if amount % (rat_1 + rat_2) == 0:
+            amt_1 = amount // (rat_1 + rat_2) * rat_1
             amt_2 = amount - amt_1
```

Comment:
This is how I computed the amounts:
```
        Double fraction = Double.valueOf(rat1) / Double.valueOf(rat1+rat2);
        int receiverOneAmount = Long.valueOf(Math.round(fraction * amount)).intValue();
        int receiverTwoAmount = Long.valueOf(Math.round((1.0 - fraction) * amount)).intValue();
```

Reply:
I think these are equivalent since amount must be divisible by (rat_1 + rat_2) because I've simplified the fraction first.

Reply:
I didn't realize `//` did an implicit floor. There might be cases where the number might need to round up after the divide though. I noticed in my code that for our test values `receiverTwoAccount` needed to round up while `receiverOneAmount` needed to round down.